src/VirtoCommerce.CustomerModule.Web/Controllers/Api/CustomerModuleController.cs

```diff
@@ -1,582 +1,668 @@
-            }
-            while (hasSearchCriteriaMembers);
-
-            listIds.ProcessWithPaging(criteria.Take, async (ids, currentItem, totalCount) =>
-            {
-                await _memberService.DeleteAsync(ids.ToArray());
-            });
-
-
-            return NoContent();
-        }
-
-        #region Special members for storefront C# API client  (because it not support polymorph types)
-
-        /// <summary>
-        /// Create contact
-        /// </summary>
-        [HttpPost]
-        [Route("contacts")]
-        [Authorize(ModuleConstants.Security.Permissions.Create)]
-        public async Task<ActionResult<Contact>> CreateContact([FromBody] Contact contact)
-        {
-            await _memberService.SaveChangesAsync(new[] { contact });
-            return Ok(contact);
-        }
-
-        /// <summary>
-        /// Bulk create contacts
-        /// </summary>
-        [HttpPost]
-        [Route("contacts/bulk")]
-        [Authorize(ModuleConstants.Security.Permissions.Create)]
-        public async Task<ActionResult<Contact[]>> BulkCreateContacts([FromBody] Contact[] contacts)
-        {
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.TagHelpers.Cache;
+using VirtoCommerce.CustomerModule.Core;
+using VirtoCommerce.CustomerModule.Core.Model;
+using VirtoCommerce.CustomerModule.Core.Model.Search;
+using VirtoCommerce.CustomerModule.Core.Services;
+using VirtoCommerce.CustomerModule.Web.Authorization;
+using VirtoCommerce.Platform.Core.Common;
+
+namespace VirtoCommerce.CustomerModule.Web.Controllers.Api
+{
+    [Route("api")]
+
+    public class CustomerModuleController : Controller
+    {
+        private readonly IAuthorizationService _authorizationService;
+        private readonly IMemberService _memberService;
+        private readonly IMemberSearchService _memberSearchService;
+
+        public CustomerModuleController(IAuthorizationService authorizationService, IMemberService memberService, IMemberSearchService memberSearchService)
+        {
+            _authorizationService = authorizationService;
+            _memberService = memberService;
+            _memberSearchService = memberSearchService;
+        }
+
+        /// <summary>
+        /// Get organizations
+        /// </summary>
+        /// <remarks>Get array of all organizations.</remarks>
+        [HttpGet]
+        [Route("members/organizations")]
+        public async Task<ActionResult<Organization[]>> ListOrganizations()
+        {
+            var searchCriteria = new MembersSearchCriteria
+            {
+                MemberType = typeof(Organization).Name,
+                DeepSearch = true,
+                Take = int.MaxValue
+            };
+            if (!(await AuthorizeAsync(searchCriteria, ModuleConstants.Security.Permissions.Access)).Succeeded)

```

Comment:
@tatarincev Please comment on this - 
Is it OK we are using Access permission for getting data auth, not Read?
It was not the case in v2, it's  new in v3.

Reply:
It's an out of scope, please discuss and do changes within separate task